client/client.go

```diff
@@ -116,6 +124,10 @@ type Client struct {
 
 	consulService *consul.ConsulService
 
+	hostStatsCollector *stats.HostStatsCollector
```

Comment:
The advantage of this being here is that if we can't create the host stats collector for some reason the client won't start. If we move this down, the client will run but we might not be able to collect stats after the client has started. And I think we shouldn't run the client if we can't create the stats collector since down the line, resource usage stats might feed into scheduling decisions.